WorkflowGuidelines.md

```diff
@@ -78,6 +78,46 @@
 
       let savedData = System.IO.File.ReadAllText saveFilePath.FullName
       ```
+    * Avoid applying logic based on exception's message:
+      
+      Exception messages (even in stable libraries/softwares) are subject to change, applying logic based on the `Message` or `StackTrace` can cause hard-to-debug problems when updating your dependencies or in case of .NET even your framework version.
```

Comment:
@aarani "in case of .NET even your framework version"? I can't parse that, please fix; also can you update this PR to follow this please: https://github.com/nblockchain/conventions/commit/b07ea57315c2cb9b11038543d0b7039b06b5e21f